### PR TITLE
[Refactor] Refactor table get mysql type

### LINKED ISSUE #18195: [StarOS] Table_type wrong from show full tables where
### Steps to reproduce the behavior (Required)
````sql
MySQL [tn_test]> SHOW FULL TABLES where Table_type!='VIEW';
+-------------------+----------------+
| Tables_in_tn_test | Table_type     |
+-------------------+----------------+
| nt2               | EXTERNAL TABLE |
| t1                | EXTERNAL TABLE |
| nt1               | EXTERNAL TABLE |
| t0                | EXTERNAL TABLE |
| t2                | EXTERNAL TABLE |
| test_lowcard      | EXTERNAL TABLE |
+-------------------+----------------+
6 rows in set (0.11 sec)

MySQL [tn_test]> SHOW FULL TABLES;
+-------------------+------------+
| Tables_in_tn_test | Table_type |
+-------------------+------------+
| nt1               | BASE TABLE |
| nt2               | BASE TABLE |
| t0                | BASE TABLE |
| t1                | BASE TABLE |
| t2                | BASE TABLE |
| test_lowcard      | BASE TABLE |
+-------------------+------------+
6 rows in set (0.00 sec)
````

### Expected behavior (Required)
BASE TABLE 

### Real behavior (Required)
EXTERNAL TABLE
from show full tables where 

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
MySQL [tn_test]> select current_version();
+-------------------+
| current_version() |
+-------------------+
| UNKNOWN 800b5477f |
+-------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by wyb**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #18195

## Problem Summary(Required) ：

1. remove duplicated function.
2. schema table returns SYSTEM VIEW.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by xiangguangyxg — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 8 (62.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/InformationSchemaDataSource.java|0|2|00.00%|[189, 296]|
|:large_blue_circle:|com/starrocks/catalog/Table.java|5|6|83.33%|[485]|

File: fe/fe-core/src/main/java/com/starrocks/catalog/Table.java [hunk 1]
```diff
@@ -459,13 +459,31 @@ public String getEngine() {
...
+            case MYSQL:
+            case BROKER:
+            case ELASTICSEARCH:
+            case HIVE:
+            case ICEBERG:
+            case HUDI:
+            case JDBC:
+            case DELTALAKE:
+            case FILE:
+                return "EXTERNAL TABLE";
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Table.java:483]: Better return base table for BI compatibility

Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/catalog/Table.java]: ok

**Review by imay — APPROVED** [review]

**Review by tracymacding — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by sduzh — APPROVED** [review]

**Comment by wyb** [comment]
run starrocks_fe_unittest

**Comment by wyb** [comment]
run starrocks_fe_unittest

**Comment by wyb** [comment]
run starrocks_fe_unittest

**Comment by wyb** [comment]
run starrocks_admit_test

**Comment by wyb** [comment]
run starrocks_admit_test

**Comment by wyb** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**